pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_advisor_handler.go

```diff
@@ -0,0 +1,300 @@
+	calculatedLimitInBytesInt64, err := strconv.ParseInt(calculatedLimitInBytes, 10, 64)
+
+	if err != nil {
+		return fmt.Errorf("parse %s: %s failed with error: %v", memoryadvisor.ControKnobKeyMemoryLimitInBytes, calculatedLimitInBytes, err)
+	}
+
+	if calculationInfo.CgroupPath != "" {
+		err = cgroupmgr.ApplyMemoryWithRelativePath(calculationInfo.CgroupPath, &cgroupcommon.MemoryData{
+			LimitInBytes: calculatedLimitInBytesInt64,
+		})
+
+		if err != nil {
+			return fmt.Errorf("apply %s: %d to cgroup: %s failed with error: %v",
+				memoryadvisor.ControKnobKeyMemoryLimitInBytes, calculatedLimitInBytesInt64,
+				calculationInfo.CgroupPath, err)
+		}
+
+		return nil
+	}
+
+	allocationInfo := podResourceEntries[v1.ResourceMemory][entryName][subEntryName]
+
+	if allocationInfo == nil {
+		return fmt.Errorf("high level cgroup path and pod resource entry are both nil")
+	} else if allocationInfo.ExtraControlKnobInfo == nil {
+		allocationInfo.ExtraControlKnobInfo = make(map[string]state.ControlKnobInfo)
+	}
+
+	allocationInfo.
+		ExtraControlKnobInfo[string(memoryadvisor.ControKnobKeyMemoryLimitInBytes)] = state.ControlKnobInfo{
+		ControlKnobValue: calculatedLimitInBytes,
+		OciPropertyName:  util.OCIPropertyNameMemoryLimitInBytes,
+	}
+
+	return nil
+}
+
+func (p *DynamicPolicy) handleAdvisorDropCache(podUID, containerName string,
+	calculationInfo *advisorsvc.CalculationInfo, podResourceEntries state.PodResourceEntries) error {
+
+	dropCache := calculationInfo.CalculationResult.Values[string(memoryadvisor.ControKnobKeyDropCache)]
+	dropCacheBool, err := strconv.ParseBool(dropCache)
+	if err != nil {
+		return fmt.Errorf("parse %s: %s failed with error: %v", memoryadvisor.ControKnobKeyDropCache, dropCache, err)
+	}
+
+	if calculationInfo.CgroupPath != "" {
+		return fmt.Errorf("dropping cache at high level cgroup path %s isn't supported",
+			memoryadvisor.ControKnobKeyDropCache)
+	}
+
+	if !dropCacheBool {
+		return nil
+	}
+
+	containerID, err := p.metaServer.GetContainerID(podUID, containerName)
+	if err != nil {
+		return fmt.Errorf("get container id of pod: %s container: %s failed with error: %v", podUID, containerName, err)
+	}
+
+	dropCacheWorkName := util.GetContainerAsyncWorkName(podUID, containerName,
+		memoryPluginAsyncWorkTopicDropCache)
+	// start a asynchronous work to drop cache for the container whose numaset changed and doesn't require numa_binding
+	err = p.asyncWorkers.AddWork(dropCacheWorkName,
+		&asyncworker.Work{
+			Fn:          cgroupmgr.DropCacheWithTimeoutForContainer,
+			Params:      []interface{}{podUID, containerID, dropCacheTimeoutSeconds},
+			DeliveredAt: time.Now()})
+
+	if err != nil {
+		return fmt.Errorf("add work: %s pod: %s container: %s failed with error: %v", dropCacheWorkName, podUID, containerName, err)
+	}
+
+	return nil
+}
+
+func (p *DynamicPolicy) handleAdvisorCPUSetMems(entryName, subEntryName string,
```

Comment:
why we use `entryName` and `subEntryName` for `handleAdvisorCPUSetMems` and `handleAdvisorMemoryLimitInBytes` instead of pod and container?